microk8s-resources/wrappers/microk8s-cilium.wrapper

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -eu
+
+export PATH="$SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH"
+export LD_LIBRARY_PATH="$SNAP/lib:$SNAP/usr/lib:$SNAP/lib/x86_64-linux-gnu:$SNAP/usr/lib/x86_64-linux-gnu"
+
+if [ ! -f "${SNAP_DATA}/bin/cilium" ]; then
+  echo "Cilium not available, try enabling is with 'microk8s.enable cilium'" >&2
+  exit 0
+fi
+
+source $SNAP/actions/common/utils.sh
+exit_if_stopped
+"${SNAP_DATA}/bin/cilium" "$@"
```

Comment:
Would you trust any local user to access cilium, or does the user need to be a sudoer?

Reply:
In typical clusters I note that the `/var/run/cilium/cilium.sock` has file mode permissions `660` with `user=root` and `group=cilium`. When I deploy Cilium via this script in microk8s right now, it's inheriting the group of the user that creates the YAMLs.

Maybe there's something I can do here to ensure that there is a cilium group on the system and that the container is run under the cilium group?

That said, access to the cilium binary allows modifying network policies for pods/endpoints, but then at the same time the same user can presumably just run `kubectl delete netpol ...` and delete policies so I don't see a real distinction there.

Reply:
Is there a way to set the path to the `cilium.sock`? We would rather not create the `/var/run/cilium/cilium.sock` in that path but instead place the socket inside $SNAP_DATA.

Reply:
Ah, I was generally under the impression that `$SNAP_DATA` isn't writable at runtime but it seems like I was just mixing it up with `$SNAP`. It looks like things like the socket path are all configurable, but will take a bit of trial and error to validate/fix them up.

Reply:
The latest snag is that the `cilium-cni` binary needs to know the cilium socket. The way that it gets this is either from the `CILIUM_SOCK` environment variable, or defaults to `unix:///var/run/cilium/cilium.sock`. I'm currently trying to see if I can make the CILIUM_SOCK environment variable available for CNI invocation via `microk8s-resources/wrappers/run-with-config-args`.

Reply:
I made a change like this for kubelet case, but this doesn't seem to work:

```
diff --git a/microk8s-resources/wrappers/run-with-config-args b/microk8s-resources/wrappers/run-with-config-args
index ea7cb0aa1131..10214288cc23 100755
--- a/microk8s-resources/wrappers/run-with-config-args
+++ b/microk8s-resources/wrappers/run-with-config-args
@@ -39,6 +39,9 @@ then
     refresh_opt_in_config "kubelet-cgroups" "/systemd/system.slice" kubelet
   fi
 
+  # Export Cilium socket path correctly for CNI
+  export CILIUM_SOCK="unix://${SNAP_DATA}/var/run/cilium/cilium.sock"
+
 elif [ "${app}" = "kube-apiserver" ] || [ "${app}" = "kube-proxy" ]
 then
   # Check if we advertise an address. If we do we do not need to wait for a default network interface.
```

Reply:
Figured it out, needs to be in `microk8s-resources/wrappers/run-containerd-with-args`. I'll update the branch shortly.